charmcraft/linters.py

```diff
@@ -197,10 +195,48 @@ def run(self, basedir: pathlib.Path) -> str:
         return result
 
 
+class JujuMetadata:
+    """Check that the metadata.yaml file exists and is sane.
+
+    The charm is considered to have a valid metadata if the following checks are true:
+
+    - the metadata.yaml is present
+    - it is a valid YAML file
+    - it has at least the following fields: name, summary, and description
+    """
+
+    check_type = CheckType.lint
+    name = "metadata"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--metadata"
+    text = "Problems found with metadata.yaml file."
```

Comment:
it would be nice to compose this, maybe text needs to be more succinct as well, like

- metadata.yaml: file missing
- metadata.yaml: not valid YAML
- metadata.yaml: missing 'name'

Reply:
I'll prepare a branch for this improvement in the future.